doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,364 @@
-                
-                import os
-                from pytorch_lightning.loggers.wandb import WandbLogger
-
-                # Grab the api key from the head node
-                WANDB_API_KEY = os.environ["WANDB_API_KEY"]
-
-                def train_func_per_worker():
-                    # Set environment variable on each worker
-                    os.environ["WANDB_API_KEY"] = WANDB_API_KEY
-                    wandb_logger = WandbLogger(...)
-                    ...
-
-                trainer = TorchTrainer(
-                    train_func_per_worker,
-                    ...
-                )
+            # This is equivalent to `os.environ["WANDB_API_KEY"] = "your_api_key"`
+            wandb.login(key="your_api_key")
+
+        **offline**
+
+        Ensure that offline directory points to a shared storage path.
+
+        .. code-block:: python
+
+            os.environ["WANDB_MODE"] = "offline"
+            wandb.init(dir="some_shared_storage_path/wandb") 
+
+    .. tab:: MLflow
         
-        .. tip::
+        **online (hosted by Databricks)**
             
-            When performing **fault-tolerant training** with auto-restoration, be sure 
-            to specify a unique ID for the Loggers, so that the new workers report to
-            the same run after restoration.
+        Ensure that all nodes have access to the Databricks config file.
+
+        .. code-block:: python
 
-            For example:
+            # MLflow client will look for a Databricks config file 
+            # at the location specified by `os.environ["DATABRICKS_CONFIG_FILE"]`.
+            os.environ["DATABRICKS_CONFIG_FILE"] = config["databricks_config_file"]
+            mlflow.set_tracking_uri("databricks")
+            mlflow.start_run()
+
+        **offline**
+
+        Ensure that offline directory points to a shared storage path.
+
+        .. code-block:: python
+
+            mlflow.start_run(tracking_uri="file:some_shared_storage_path/mlruns")
+
+Setting up credentials
+~~~~~~~~~~~~~~~~~~~~~~
+
+Please refer to each tracking library's API documentation on this.
+This usually involves setting some environment variable or accessing some config file.
+
+For setting environment variable, one may pass the value of environment variable through the ``config`` 
+argument of ``train_func`` and set the corresponding environment variable in the ``train_func``.
+
+For accessing config file, one needs to ensure that the config file is accessible to all nodes.
+One way to do this is by setting up a shared storage. Another way is to save a copy in each node.
+
+Setting up shared file system
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This involves setting up a network filesystem accessible to all nodes in the cluster, 
+e.g. AWS EFS or Google Cloud Filestore.
+
+Step 2: Configure and start the run 
+-----------------------------------
+
+This usually concerns picking an identifier for the run and associating it with a project.
```

Comment:
One thing we discussed yesterday was about how to link this experiment with the Ray Train results folder/metadata. You mentioned that you didn't want to use the trial name, but would prefer to recommend logging the metadata as part of the config instead. Can you add that?